### ISSUE TITLE
terminating with uncaught exception of type realm SyntaxError: JSON Parse error: Unexpected identifier "undefined"

### ISSUE THREAD

**Issue by mMarcos208** [realm/realm-js#3683]
## Goals
Delete all data

## Expected Results
Delete all data

## Actual Results
Application Specific Information:
terminating with uncaught exception of type realm::js::Exception<realm::jsc::Types>: SyntaxError: JSON Parse error: Unexpected identifier "undefined"
abort() called
CoreSimulator 732.18.6 - Device: iPhone 11 (00896F50-4B8B-4205-8FB6-BFE7687B53F0) - Runtime: iOS 14.4 (18D46) - DeviceType: iPhone 11

## Steps to Reproduce
* Close my app
* Open my app
* Remove all data of schemas

## Code Sample
```
   if (shouldRemoveUserData) {
      realm.delete(realm.objects(A.schema.name))
      realm.delete(realm.objects(B.schema.name))
      realm.delete(realm.objects(C.schema.name))
    }
    realm.delete(realm.objects(D.schema.name))
    realm.delete(realm.objects(E.schema.name))
    realm.delete(realm.objects(F.schema.name))
    realm.delete(realm.objects(G.schema.name))
    realm.delete(realm.objects(H.schema.name))
    realm.delete(realm.objects(I.schema.name))
    realm.delete(realm.objects(J.schema.name))
    realm.delete(realm.objects(L.schema.name))
    realm.delete(realm.objects(M.schema.name))
    realm.delete(realm.objects(N.schema.name))
    realm.delete(realm.objects(O.schema.name))
    realm.delete(realm.objects(P.schema.name))
```

## Version of Realm and Tooling
* Realm JS SDK Version: ? 10.2.0
* Node or React Native: ? v15.0.1
* Client OS & Version: ? Mac Os Big Sur
* Which debugger for React Native: ? NONE

**Comment by kneth** [comment]
Are there any [relationships](https://docs.mongodb.com/realm/sdk/react-native/fundamentals/relationships-and-embedded-objects/#std-label-react-native-client-relationships) in your schema? It should matter but I am trying to narrow down the possibilities (and see where the bug is).

From the error message (`Device iPhone`) my guess is that it is a React Native app. Do you have a debugger attached? If yes, so you see the same error if the debugger is not attached?

**Comment by mMarcos208** [comment]
@kneth Yes, I've relationships in my schema.

I should delete my relation firts?

**From the error message (Device iPhone) my guess is that it is a React Native app. Do you have a debugger attached? If yes, so you see the same error if the debugger is not attached?**

Yes, in my console.

**Comment by kneth** [comment]
@mMarcos208 

> Should I delete my relation firts?

It could be interesting to know if it makes a difference. It doesn't matter but we might have an edge case that we don't cover correctly.

> Yes, in my console

Please try to disable it. It will help us to understand if it is a database issue (the above) or a debugger issue.

**Comment by mMarcos208** [comment]
I disabled my debugger.

This error happen when I close my app and open again.

**Comment by kneth** [comment]
We have released v11.0.0 which has a complete rewrite of the integration with JSC. I am closing the issue, and ask you to upgrade. If the issue is still observed, please create a new issue.

**Comment by mMarcos208** [comment]
@kneth thank's, the lib is is working well.
